jgrapht-guava/src/main/java/org/jgrapht/graph/guava/BaseNetworkAdapter.java

```diff
@@ -74,10 +77,28 @@ public BaseNetworkAdapter(N network)
      * @param edgeSupplier the edge supplier
      */
     public BaseNetworkAdapter(N network, Supplier<V> vertexSupplier, Supplier<E> edgeSupplier)
```

Comment:
The Graph interface in Guava exposes a nodeOrder method.  We could call nodeOrder().type(), and if it's not UNORDERED, then we can rely on a guaranteed order already being available and use that.

Reply:
They only provide a `Comparator` in case of `SORTED`. The other two orders (`INSERTION`, `STABLE`) are implicit inside the guava graph.